ttnn/cpp/ttnn/operations/moreh/moreh_linear_backward/device/moreh_linear_backward_device_operation.cpp

```diff
@@ -0,0 +1,115 @@
+// SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "moreh_linear_backward_device_operation.hpp"
+
+#include <cstdint>
+
+#include "tt_dnn/op_library/moreh_helper_functions.hpp"
+#include "ttnn/operations/core/compute_kernel/compute_kernel_config.hpp"
+#include "ttnn/tensor/tensor.hpp"
+
+namespace ttnn::operations::moreh::moreh_linear_backward {
+
+void MorehBiasAddBackwardOperation::validate_inputs(
+    const operation_attributes_t& operation_attributes, const tensor_args_t& tensor_args) {
+    auto& output_grad = tensor_args.output_grad;
+    auto& input = tensor_args.input;
+    auto& weight = tensor_args.weight;
+    auto& input_grad = tensor_args.input_grad;
+    auto& weight_grad = tensor_args.weight_grad;
+    auto& bias_grad = tensor_args.bias_grad;
+
+    if (input_grad.has_value()) {
+        auto input_shape = input.get_shape();
+        auto input_grad_shape = input_grad->get_shape();
+        TT_ASSERT(input_shape == input_grad_shape, "both tensors should be the same shape");
```

Comment:
```suggestion
        TT_FATAL(input_shape == input_grad_shape, "both tensors should be the same shape");
```

Reply:
resolved in https://github.com/tenstorrent/tt-metal/pull/12812/commits/8dec1ff696bd9948f9366cbee8ec5733a58f2081